dashboard/app/controllers/omniauth_callbacks_controller.rb

```diff
@@ -81,12 +81,13 @@ def login
       redirect_to '/home?open=rosterDialog'
     elsif User::OAUTH_PROVIDERS_UNTRUSTED_EMAIL.include?(provider) && @user.persisted?
       handle_untrusted_email_signin(@user, provider)
+    elsif allows_silent_takeover(@user, auth_hash) || google_classroom_student_takeover(@user)
```

Comment:
What's the need to move this above the persisted case? I see that you added a guard in the `allows_silent_takeover` method, so it should be safe, I'm just curious.

Reply:
we talked in slack, but also want to document here: because a student who meets the requirements for `google_classroom_student_takeover` will always also be persisted, we have to move the silent takeover check before the `elsif @user.persisted?` check